From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 12 Apr 2017 14:08:38 +0200
Subject: clk: mvebu: mark root clocks in AP806/CP110 system-controller drivers
Patch-mainline: Never, we backport CLK_IS_ROOT to make the driver work on v4.4 based kernels
References: bsc#1032150

Clk infrastructure in LKv4.4.x require registering root clocks in
the clk tree with CLK_IS_ROOT flag. Otherwise obtaining parameters
of child clocks will fail.

This patch adds CLK_IS_ROOT flag for according AP806/CP110 clocks.

Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/clk/mvebu/ap806-system-controller.c | 8 ++++----
 drivers/clk/mvebu/cp110-system-controller.c | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index f17702107ac5..5b044f3d1b8d 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -100,7 +100,7 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 	of_property_read_string_index(np, "clock-output-names",
 				      0, &name);
 	ap806_clks[0] = clk_register_fixed_rate(&pdev->dev, name, NULL,
-						0, cpuclk_freq);
+						CLK_IS_ROOT, cpuclk_freq);
 	if (IS_ERR(ap806_clks[0])) {
 		ret = PTR_ERR(ap806_clks[0]);
 		goto fail0;
@@ -108,8 +108,8 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 
 	of_property_read_string_index(np, "clock-output-names",
 				      1, &name);
-	ap806_clks[1] = clk_register_fixed_rate(&pdev->dev, name, NULL, 0,
-						cpuclk_freq);
+	ap806_clks[1] = clk_register_fixed_rate(&pdev->dev, name, NULL,
+						CLK_IS_ROOT, cpuclk_freq);
 	if (IS_ERR(ap806_clks[1])) {
 		ret = PTR_ERR(ap806_clks[1]);
 		goto fail1;
@@ -119,7 +119,7 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 	of_property_read_string_index(np, "clock-output-names",
 				      2, &fixedclk_name);
 	ap806_clks[2] = clk_register_fixed_rate(&pdev->dev, fixedclk_name, NULL,
-						0, 1200 * 1000 * 1000);
+						CLK_IS_ROOT, 1200 * 1000 * 1000);
 	if (IS_ERR(ap806_clks[2])) {
 		ret = PTR_ERR(ap806_clks[2]);
 		goto fail2;
diff --git a/drivers/clk/mvebu/cp110-system-controller.c b/drivers/clk/mvebu/cp110-system-controller.c
index 6b11d7b3e0e0..0bcb4c4eda88 100644
--- a/drivers/clk/mvebu/cp110-system-controller.c
+++ b/drivers/clk/mvebu/cp110-system-controller.c
@@ -217,7 +217,7 @@ static int cp110_syscon_clk_probe(struct platform_device *pdev)
 	/* Register the APLL which is the root of the hw tree */
 	of_property_read_string_index(np, "core-clock-output-names",
 				      CP110_CORE_APLL, &apll_name);
-	hw = clk_hw_register_fixed_rate(NULL, apll_name, NULL, 0,
+	hw = clk_hw_register_fixed_rate(NULL, apll_name, NULL, CLK_IS_ROOT,
 					1000 * 1000 * 1000);
 	if (IS_ERR(hw)) {
 		ret = PTR_ERR(hw);
-- 
2.12.0

